plan/triggers.go

```diff
@@ -41,6 +45,84 @@ func SetTriggerSpec(node PlanNode) error {
 	return nil
 }
 
+// OptimizeWindows optimizes the triggering mechanism for certain window transformations.
+// In particular, if a windowing operation occurs in a query plan and does not have any
+// dependencies on other window operations, then this particular window operation may send
+// its output tables immediately downstream after processing an input table.
+func OptimizeWindows(spec *PlanSpec) error {
```

Comment:
Is there any reason this can't fit into the rule framework?  There will be lots of optimizations like this, and I'd prefer to extend the rule framework to allow for this kind of transformation, rather than do many passes like this.

Reply:
@wolffcm thank you for the suggestion. The window optimization has been changed to be a physical plan rule. It's much more concise. The logic is in the rule itself now, not in the planner.